docset/winserver2022-ps/netsecurity/New-NetFirewallDynamicKeywordAddress.md

```diff
@@ -0,0 +1,212 @@
+schema: 2.0.0
+title: New-NetFirewallDynamicKeywordAddress
+---
+
+# New-NetFirewallDynamicKeywordAddress
+
+## SYNOPSIS
+Creates a dynamic keyword address.
+
+## SYNTAX
+
+```
+New-NetFirewallDynamicKeywordAddress [-Id <String>] [-Keyword <String>] [-Addresses <String>]
+ [-AutoResolve <Boolean>] [-CimSession <CimSession[]>] [-ThrottleLimit <Int32>] [-AsJob] [-WhatIf] [-Confirm]
+ [<CommonParameters>]
+```
+
+## DESCRIPTION
+The **New-NetFirewallDynamicKeywordAddress** cmdlet creates a dynamic keyword address.
+
+A firewall rule can use dynamic keyword addresses instead of explicitly defining IP addresses for its remote address condition.
+
+## EXAMPLES
+
+### Example 1: Create dynamic keyword address with AutoResolve
+```powershell
+ New-NetFirewallDynamicKeywordAddress -Id 258741ed-94a3-406b-9b0c-81fb145a4592 -Keyword ContosoSubnet -AutoResolve
+```
+
+This command creates an AutoResolve dynamic keyword address.
+
+Be sure to generate your own GUID.
+
+### Example 2: Create dynamic keyword address
+```powershell
+ New-NetFirewallDynamicKeywordAddress -Id b2a86fcb-44c8-4527-94b2-e911aba257d2 -Keyword ContosoServerSubnet -Addresses 10.0.0.21
+```
+
+This command creates a non-AutoResolve dynamic keyword address.
+
+Be sure to generate your own GUID.
+
+## PARAMETERS
+
+### -Addresses
+Specifies the addresses for the dynamic keyword address.
+
+```yaml
+Type: String
+Parameter Sets: (All)
+Aliases:
+
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -AsJob
+Runs the cmdlet as a background job. Use this parameter to run commands that take a long time to complete.
+
+```yaml
+Type: SwitchParameter
+Parameter Sets: (All)
+Aliases:
+
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -AutoResolve
+Whether the dynamic key address uses AutoResolve.
+For AutoResolve, the keyword field represents a resolvable name and the IP addresses aren't defined when the dynamic key address is created.
+
+```yaml
+Type: Boolean
+Parameter Sets: (All)
+Aliases:
+
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -CimSession
+Runs the cmdlet in a remote session or on a remote computer.
+Enter a computer name or a session object, such as the output of a [New-CimSession](https://go.microsoft.com/fwlink/p/?LinkId=227967) or [Get-CimSession](https://go.microsoft.com/fwlink/p/?LinkId=227966) cmdlet.
+The default is the current session on the local computer.
+
+```yaml
+Type: CimSession[]
+Parameter Sets: (All)
+Aliases: Session
+
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -Id
+Specifies a GUID for the dynamic keyword address.
+If you do not specify a GUID, the cmdlet creates one.
```

Comment:
Ah, ok, cool. We should repeat this in the examples where we tell the reader to generate their own GUID. That section and this section give conflicting advice currently, and I think we should standardize on the easy behavior which is "leave the GUID generation to us" since GUIDs are inherently not human friendly and the reader is unlikely to have a GUID they prefer to another.

Reply:
I'm pulling the second line. You verified using a GUID in the example, so I'll go with that. I don't want to go beyond what I'm sure of.